tests/unit/test_paramfile.py

```diff
@@ -10,14 +10,99 @@
 # distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
 # ANY KIND, either express or implied. See the License for the specific
 # language governing permissions and limitations under the License.
-import platform
-
 import mock
 from awscli.compat import six
 from awscli.testutils import unittest, FileCreator
 from awscli.testutils import skip_if_windows
 
 from awscli.paramfile import get_paramfile, ResourceLoadingError
+from awscli.paramfile import UriArgumentHandler
+from awscli.paramfile import LOCAL_PREFIX_MAP, REMOTE_PREFIX_MAP
+from awscli.paramfile import register_uri_param_handler
+from botocore.hooks import HierarchicalEmitter
+
+
+class FakeSession(object):
+    def __init__(self, config, emitter=None):
+        self.config = config
+        if emitter is None:
+            emitter = HierarchicalEmitter()
+        self.emitter = emitter
+
+    def register(self, event_name, handler):
+        self.emitter.register(event_name, handler)
+
+    def emit(self, event_name, **kwargs):
+        return self.emitter.emit(event_name, **kwargs)
+
+    def get_scoped_config(self):
+        return self.config
+
+
+class TestUriArgumentHandler(unittest.TestCase):
+    def test_default_does_have_local_and_remote(self):
+        with mock.patch(
+                'awscli.paramfile.get_paramfile') as mock_get_paramfile:
+            handler = UriArgumentHandler()
+            handler('event-name', 'param', 'value')
+            prefixes = mock_get_paramfile.call_args[0][1]
+
+        self.assertEqual(len(prefixes), 4)
+        self.assertIn('file://', prefixes)
+        self.assertIn('fileb://', prefixes)
+        self.assertIn('http://', prefixes)
+        self.assertIn('https://', prefixes)
+
+    def test_uses_supplied_cases(self):
+        with mock.patch(
```

Comment:
It would be great if we can minimize the use of patches. Specifically, it would be nice if we provide our own recording handler (or something like that) that we ensure gets called. This concerns are a little less for the default scenarios, but it may make sense to instead to switch some of these to functional tests (i.e. make sure each of the prefix types work).